src/analyzing/AST_to_IL.ml

```diff
@@ -1047,7 +1047,11 @@ and stmt_expr env ?e_gen st =
     | Some e_gen -> todo (G.E e_gen)
   in
   match st.G.s with
-  | G.ExprStmt (eorig, _) -> expr env eorig
+  | G.ExprStmt (eorig, tok) ->
+      (if eorig.is_implicit_return then
+       let ss, e = expr_with_pre_stmts ~void:true env eorig in
+       ss @ [ mk_s (Return (tok, e)) ] |> add_stmts env);
+      expr env eorig
```

Comment:
Here shouldn't just return `e` ? Why translating `eorig` twice?

Reply:
Ah, yes, it should be translated once. Thanks.